### DISCUSSION TITLE
Control / Set the `fragment-index` of `{.incremental}` Explicitly

### DISCUSSION THREAD

**Discussion by RoyiAvital** [quarto-dev/quarto-cli#3559] [Feature Requests]
Assume the following:

```md
## How to Measure

::: {.incremental}
- Assume we have 2 variants of a website: `A` and `B`.
- How would we measure their design effect on user engagement?
:::

![](https://i.imgur.com/PSZggyN.png){.absolute bottom=0 .fragment .fade-in-then-out fragment-index=1}
```

It would be useful to be able to control the `fragment-index` of the items in the list in order to sync them with other items in the slide.  
Of course the need is only to set the index of the first item in the list.

**Remark**: It was partly discussed in https://github.com/quarto-dev/quarto-cli/issues/3539.

**Comment by RoyiAvital** [comment]
Is there an alternative way to have a list of items with fragments with explicit `fragment-index=1`?

**Reply by mcanouil** [reply]
yes, see other more recent discussions such as:

- https://github.com/orgs/quarto-dev/discussions/13263

**Reply by RoyiAvital** [reply]
I understand what you did. Which is a great solution.

I wonder if one can create a filter or other script which will designate specific indices.

**Reply by mcanouil** [reply]
That's definitely doable and that's on my to-do list.
I don't know if this is going to be an extension or a PR to Quarto.

I have already several Reveal.js plugins for workarounds for fragments related components (https://github.com/mcanouil/quarto-mcanouil/tree/main/_extensions/mcanouil/revealjs-plugin)
- tabset (direct use of the plugin https://github.com/quarto-dev/quarto-cli/pull/13752)
- code-annotation
- code line indices (a more direct support via https://github.com/quarto-dev/quarto-cli/pull/13125)

**Reply by RoyiAvital** [reply]
That's great. Your contribution to the eco system is amazing, thanks for that.

**Comment by mcanouil** [comment]
Here is "fragment ion": https://github.com/mcanouil/quarto-revealjs-fragmention

![comparison between native span for fragment in list versus "fragmention"](https://github.com/user-attachments/assets/10867921-1a00-4674-a5ef-4e531c4ca70e)

**Reply by RoyiAvital** [reply]
Great! Thanks for the effort.
